### PR TITLE
Add general facility for arbitrary fluxmonitor threshold functions

### PR DESCRIPTION
**Pull request by coventry**
This adds an absoluteThreshold option, in addition to the extant relativeThreshold.

### CODE REVIEW

File: core/internal/testdata/flux_monitor_job.json [hunk 1]
```diff
@@ -1,4 +1,12 @@
 {
+  "comment_comments": [
```
Comment (se3000) [inline core/internal/testdata/flux_monitor_job.json:2]: I think these comments would be more discoverable in a package than in a test fixture.

File: core/services/validators.go [hunk 2]
```diff
@@ -128,12 +128,12 @@ func validateFluxMonitor(i models.Initiator, j models.JobSpec, store *store.Stor
 	if i.Address == utils.ZeroAddress {
 		fe.Add("no address")
 	}
+	if len(i.ValueTriggers) == 0 {
+		fe.Add("must have at least one value trigger")
```
Comment (se3000) [inline core/services/validators.go:132]: I believe we want to support only idleThreshold. @thodges-gh or @spooktheducks ?

File: core/store/migrations/migration1587591248/migration.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+)
+
+func Migrate(tx *gorm.DB) error {
+	err := tx.Exec(`
+ALTER TABLE initiators ADD COLUMN "value_triggers" text;
+ALTER TABLE initiators DROP COLUMN "threshold"
+	`).Error
+	return errors.Wrapf(err, "while migrating initiators to include "+
+		"value_triggers column")
+	// XXX: Add actual migration, here
```
[the remark this answers is not in the document]
Comment (coventry) [reply core/store/migrations/migration1587591248/migration.go:15]: We will need an actual migration for existing fluxmonitor-initiated jobs, though, right?

File: core/store/migrations/migration1587591248/migration.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package migration1587591248
+
+import (
+	"github.com/jinzhu/gorm"
+	"github.com/pkg/errors"
+)
+
+func Migrate(tx *gorm.DB) error {
+	err := tx.Exec(`
+ALTER TABLE initiators ADD COLUMN "value_triggers" text;
```
[the remark this answers is not in the document]
Comment (coventry) [reply core/store/migrations/migration1587591248/migration.go:10]: sure.

File: core/internal/testdata/flux_monitor_job.json (hunk 1 above)
Comment (coventry) [reply core/internal/testdata/flux_monitor_job.json:2]: I can remove them, but FWIW, the job specs were one of the first places I looked when I was figuring chainlink out, and they were confusing.

File: core/services/validators.go (hunk 2 above)
Comment (coventry) [reply core/services/validators.go:132]: I will add a trivial value trigger, which always triggers. I would rather require the user be explicit about that than make it the default.

File: core/store/migrations/migration1587591248/migration.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (samsondav) [reply core/store/migrations/migration1587591248/migration.go:15]: Doesn't matter does it? There are none out there in the wild afaik.

**Comment by coventry** [comment]
As discussed with @se3000  and @spooktheducks today, closing this in favor of a simpler hard-coded absolute threshold for now.

**Pull request closed without merging**